cmd/run.go

```diff
@@ -214,49 +214,87 @@ func createAggSenderValidator(ctx context.Context,
 	l2Syncer *bridgesync.BridgeSync,
 	l1Client aggkittypes.BaseEthereumClienter,
 	rollupDataQuerier *etherman.RollupDataQuerier) (*aggsender.AggsenderValidator, error) {
-	logger := log.WithFields("module", aggkitcommon.AGGSENDERVALIDATOR)
-
-	signer, err := signer.NewSigner(ctx, 0, cfg.Signer, aggkitcommon.AGGSENDERVALIDATOR, logger)
-	if err != nil {
-		return nil, fmt.Errorf("error NewSigner. Err: %w", err)
-	}
-	if err := signer.Initialize(ctx); err != nil {
-		return nil, fmt.Errorf("error signer.Initialize. Err: %w", err)
+	if err := cfg.Validate(); err != nil {
+		return nil, fmt.Errorf("invalid aggsender validator config: %w", err)
 	}
 
-	l2BridgeQuerier := query.NewBridgeDataQuerier(logger, l2Syncer, cfg.DelayBetweenRetries.Duration)
-	l1InfoTreeQuerier := query.NewL1InfoTreeDataQuerier(l1Client, l1InfoTreeSync)
-	lerQuerier := query.NewLERDataQuerier(cfg.LerQuerier.RollupCreationBlockL1, rollupDataQuerier)
+	logger := log.WithFields("module", aggkitcommon.AGGSENDERVALIDATOR)
 	agglayerClient, err := agglayer.NewAgglayerClient(cfg.AgglayerClient)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create agglayer grpc client: %w", err)
 	}
-	flowPP := flows.NewPPFlow(
-		logger,
-		flows.NewBaseFlow(
-			logger,
-			l2BridgeQuerier,
-			nil, // storage
-			l1InfoTreeQuerier,
-			lerQuerier,
-			flows.BaseFlowConfig{
-				MaxCertSize:          cfg.MaxCertSize,
-				StartL2Block:         0,
-				RequireNoFEPBlockGap: false, //  for PP doesnt apply it
-			},
-		),
-		nil, // storage
-		l1InfoTreeQuerier,
-		l2BridgeQuerier,
-		signer, // we reuse the signer, but the PP signature is not use
-		cfg.RequireOneBridgeInPPCertificate,
-		cfg.MaxL2BlockNumber,
+
+	var (
+		flow                 validator.FlowInterface
+		commonFlowComponents *flows.CommonFlowComponents
 	)
 
+	switch cfg.Mode {
+	case aggsendertypes.PessimisticProofMode.String():
+		commonFlowComponents, err = flows.CreateCommonFlowComponents(
+			ctx, logger,
+			nil, // storage is not used in validator,
+			l1Client, l1InfoTreeSync, l2Syncer, rollupDataQuerier, 0, false,
+			cfg.MaxCertSize, cfg.LerQuerier.RollupCreationBlockL1, cfg.DelayBetweenRetries.Duration, cfg.Signer,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create common flow components: %w", err)
+		}
+
+		flow = flows.NewPPFlow(
+			logger,
+			flows.NewBaseFlow(
+				logger,
+				commonFlowComponents.L2BridgeQuerier,
+				nil, // storage is not used in validator
+				commonFlowComponents.L1InfoTreeDataQuerier,
+				commonFlowComponents.LERQuerier,
+				flows.NewBaseFlowConfig(cfg.MaxCertSize, 0, false),
+			),
+			nil, // storage is not used in validator
```

Comment:
Shall we create some no op storage impl (just in case), wdyt?

Reply:
I would leave it as nil. If it uses storage, it should fail, and we should see this. It means we messed something up.